### PR TITLE
Stream proposals from mmf to synchronizer

### PR DESCRIPTION
**Pull request by Laremere**
This improves efficiency for overall system latency, and sets up for better mmf error handling.

The overall structure of the fetch matches call has been reworked.  The different go routines now set an explicit err variable.  So once we have FetchSummary, we can just set the mmf err variable on it.  Synchronizer calls which err will always result in an error here (as it's relatively fatal), while mmf and evaluator errors will be passed gently to the client.

One thing this code isn't doing anymore is checking if an mmf returns a match with no tickets.  This seems fine to me, but willing to discuss if anyone disagrees.

Deleted the tests for the following reasons:
- TestDoFetchMatchesInChannel didn't actually test fetching matches, it only tested creating a client.  Since callMmf now both creates the client and makes the call, this code now blocks actually trying to make a connection.  I'm not worried about having full branch test coverage on err statements...
- TestDoFetchMatchesFilterChannel tested merging of mmf runs.  Since there's only one mmf run now, it's no longer necessary.

### CODE REVIEW

File: internal/app/backend/backend_service.go [hunk 1]
```diff
@@ -75,198 +70,194 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
...
+
+	// TODO: Send mmf error in FetchSummary instead of erroring call.
+	if synchronizeSendErr != nil || synchronizeRecvErr != nil || mmfErr != nil {
+		logger.WithFields(logrus.Fields{
+			"synchronizeSendErr": synchronizeSendErr,
+			"synchronizeRecvErr": synchronizeRecvErr,
+			"mmfErr":             mmfErr,
+		}).Error("error(s) in FetchMatches call.")
+
+		return fmt.Errorf(
```
Comment (yfei1) [inline internal/app/backend/backend_service.go]: Would it make more sense to use `status.Errorf` at here?

File: internal/app/backend/backend_service.go [hunk 2]
```diff
@@ -75,198 +70,194 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
...
-	for i := 0; i < 2; i++ {
-		err := <-errors
-		if err != nil {
-			logger.WithFields(logrus.Fields{
-				"error": err.Error(),
-			}).Error("error in FetchMatches call.")
-			return err
+		if resp.Match != nil {
+			telemetry.RecordUnitMeasurement(stream.Context(), mMatchesFetched)
+			err = stream.Send(&pb.FetchMatchesResponse{Match: resp.Match})
```
Comment (yfei1) [inline internal/app/backend/backend_service.go]: nit: resp.GetMatch()

File: internal/app/backend/backend_service.go (hunk 1 above)
Comment (Laremere) [reply internal/app/backend/backend_service.go]: The way I'm combining errors, not really.  It's not clear, eg, if there are different status codes what the combined one should be.  FetchSummary will alleviate this and this call shouldn't ever end with an error being returned.

File: internal/app/backend/backend_service.go (hunk 2 above)
Comment (Laremere) [reply internal/app/backend/backend_service.go]: err is nil, so resp is not nil.  There's also the above line resp.Match != nil, which guarantees there to be a value here.

Comment (yfei1) [reply internal/app/backend/backend_service.go]: I understand the code logic, but I am more concerned about using the proto getters as a standard in our repo.

**Pull request merged**